playbooks/roles/grafana/tasks/main.yml

```diff
@@ -0,0 +1,113 @@
+---
+- name: Include variables
+  include_vars: "{{ lookup('first_found', params) }}"
+  vars:
+    params:
+      files: "{{ distro_lookup_path }}"
+      paths:
+        - "vars"
+
+- name: Install required packages
+  become: true
+  ansible.builtin.package:
+    state: present
+    name: "{{ item }}"
+  loop:
+    - "{{ packages }}"
+  when: "ansible_facts.pkg_mgr != 'atomic_container'"
+  register: task_result
+  until: task_result is success
+  retries: 5
+
+- name: Create Grafana group
+  become: true
+  ansible.builtin.group:
+    name: "{{ grafana_os_user }}"
+    state: present
+
+- name: Create Grafana user for docker/podman
+  become: true
+  ansible.builtin.user:
+    name: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    groups: "{{ 'docker' if container_command == 'docker' else '' }}"
+    state: present
+
+- name: Create required directories
+  become: true
+  file:
+    path: "{{ item.dest }}"
+    state: directory
+    owner: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    mode: "{{ item.mode | default(omit) }}"
+  loop:
+    - {dest: "/etc/grafana"}
+    - {dest: "/etc/grafana/provisioning"}
+    - {dest: "/etc/grafana/ldap"}
+
+- name: Write Grafana config env file
+  become: true
+  template:
+    src: "env.j2"
+    dest: "/etc/grafana/env"
+    owner: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    mode: 0664
+
+- name: Write Grafana ldap conf file
+  become: true
+  template:
+    src: "ldap.toml.j2"
+    dest: "/etc/grafana/ldap/ldap.toml"
+    owner: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    mode: 0664
+  when: grafana_auth_ldap_enable
+
+- name: Write Grafana ldap root CA file
+  become: true
+  copy:
+    content: "{{ grafana_auth_ldap_certificate }}"
+    dest: "/etc/grafana/ldap/ldap-certificate.crt"
+    owner: "{{ grafana_os_user }}"
+    group: "{{ grafana_os_group }}"
+    mode: 0664
+  when: grafana_auth_ldap_enable
+
+- include_tasks: firewall.yml
```

Comment:
this can be dropped once general firewall management role is used

Reply:
dropped